pkg/cluster/scaler/scaler.go

```diff
@@ -12,6 +18,22 @@ import (
 	"github.com/openshift/openshift-azure/pkg/util/azureclient"
 )
 
+type Factory interface {
+	New(log *logrus.Entry, ssc azureclient.VirtualMachineScaleSetsClient, vmc azureclient.VirtualMachineScaleSetVMsClient, kubeclient kubeclient.Kubeclient, resourceGroup string, ss *compute.VirtualMachineScaleSet) Scaler
+}
+
+type scalerFactory struct{}
+
+func NewFactory() Factory {
+	return &scalerFactory{}
+}
+
+type Scaler interface {
+	Scale(ctx context.Context, count int64) *api.PluginError
```

Comment:
Just by looking at this interface it's not clear to me what's the point of Scale when there is ScaleUp and ScaleDown. Certainly comments would help if this is still required.

Reply:
Hrm, Scale seems to encapsulate ScaleUp and ScaleDown. I don't think this is the right interface to expose to clients :thinking:

Reply:
@kargakis I am just exposing the function that are used here:
https://github.com/openshift/openshift-azure/pull/1027/files#diff-848159614a4ba38049bdf602a8654627R100

Reply:
would be useful to add docs since `count` is used differently in Scale than ScaleUp and ScaleDown.

Reply:
will do